src/Akka.Persistence.Hosting/AkkaPersistenceHostingExtensions.cs

```diff
@@ -156,6 +187,79 @@ public static AkkaConfigurationBuilder WithJournalAndSnapshot(
             return builder.AddHocon(defaultConfig, HoconAddMode.Append);
         }
 
+        /// <summary>
+        /// A generic way to add both journal and snapshot store configuration to the <see cref="ActorSystem"/> with support for event adapters and health checks.
+        /// </summary>
+        /// <param name="builder">The builder instance being configured.</param>
+        /// <param name="journalOptions">The specific journal options instance used to configure the journal. For example, an instance of <c>SqlServerJournalOptions</c></param>
+        /// <param name="snapshotOptions">The specific snapshot store options instance used to configure the snapshot store. For example, an instance of <c>SqlServerSnapshotOptions</c></param>
+        /// <param name="configureJournal">Optional action to configure event adapters and health checks for the journal.</param>
+        /// <param name="configureSnapshot">Optional action to configure health checks for the snapshot store.</param>
+        /// <returns>The same <see cref="AkkaConfigurationBuilder"/> instance originally passed in.</returns>
+        /// <exception cref="ArgumentNullException"></exception>
+        /// <example>
+        /// <code>
+        /// builder.WithJournalAndSnapshot(
+        ///     new SqlServerJournalOptions
+        ///     {
+        ///         ConnectionString = "...",
+        ///         IsDefaultPlugin = true
+        ///     },
+        ///     new SqlServerSnapshotOptions
+        ///     {
+        ///         ConnectionString = "...",
+        ///         IsDefaultPlugin = true
+        ///     },
+        ///     journal => journal
+        ///         .AddWriteEventAdapter&lt;MyAdapter&gt;("adapter", new[] { typeof(MyEvent) })
+        ///         .WithHealthCheck(HealthStatus.Degraded),
+        ///     snapshot => snapshot
+        ///         .WithHealthCheck());
+        /// </code>
+        /// </example>
+        public static AkkaConfigurationBuilder WithJournalAndSnapshot(
```

Comment:
DX improvement: the `JournalOptions` and `JournalBuilder` should be part of the same "configuration" - the way it's defined in v1.5.50 right now is as two separate islands of configuration that reference the same thing via a shared pluginId.

We should, instead, make it clear to end-users that both of these components work against the same "thing".

We have also followed this pattern for the snapshot store too.